submissions/ArthurGorbenko/frogger-game/js/app.js

```diff
@@ -0,0 +1,126 @@
+
+let countForWonRound = 0;
+
+const blockWidth = 100;
+const blockHeight = 90;
+const fieldMaxHeight = 400;
+const fieldMinHeight = -70;
+const fieldMaxWidth = 405;
+const fieldMinWidth = 0;
+const waterLevel = fieldMinHeight + blockHeight;
+const startPositionX = 200;
+const startPositionY = 300;
+
+class Enemy {
+    constructor(x, y) {
+        this.sprite = 'images/enemy-bug.png';
+        this.x = x;
+        this.y = y;
+        this.speed = Math.floor((Math.random() * 200)) + 100;
+    }
+    update(dt) {
+        this.x += this.speed * dt;
+        
+        if (this.x > 600) {
+            this.x = -100;
+        }
+
+        const halfOfSizeImg = 50;
+
+        if (this.x < player.x + halfOfSizeImg &&
+            this.x + halfOfSizeImg > player.x &&
+            this.y < player.y + halfOfSizeImg &&
+            this.y + halfOfSizeImg > player.y) {
+            alert('You are busted!')
+            countForWonRound = 0;
+            player.x = startPositionX;
+            player.y = startPositionY;
+            allEnemies.forEach(el => el.generateSpeed());
+        }
+    }
+    render() {
+        ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+    }
+    generateSpeed(){
+        this.speed = Math.floor((Math.random() * 200)) + 100;
```

Comment:
Every round enemy speed based on the previous round or initial speed. On the 4th round, speed is too high

https://youtu.be/m1HkvR993Go

Reply:
Did it in one commit with [speed changing](https://github.com/kottans/frontend-2019-p2p/pull/79/commits/634a323f524ae659d14923592dbeaf883c82fbaf).